### ISSUE TITLE
api3/network: Crash when a player quits due to refusing resource packs

### ISSUE THREAD

**Issue by dktapps** [pmmp/PocketMine-MP#471]
### Issue description
The entity constructor has not yet been called when the player is closed, but their name has been set. This causes a crash when saving player data when the player disconnects due to refusing resource packs.

Relates to #374 

### Steps to reproduce the issue

1. force resources
2. connect, refuse resources

### OS and versions

* PocketMine-MP: 52f2596dc56f654673f0e727090fa621847f536f
* PHP: 7.0.17
* Server OS: w10
* Game version: not relevant

### Crashdump, backtrace or other files
```
[16:07:46] [Server thread/CRITICAL]: Error: "Call to a member function getAttribute() on null" (EXCEPTION) in "C:/Users/User/Documents/GitHub/pocketmine-mp/src/pocketmine/entity/Human" at line 104
[16:07:46] [Server thread/DEBUG]: #0 C:/Users/User/Documents/GitHub/pocketmine-mp/src/pocketmine/entity/Human(412): pocketmine\entity\Human->getFood(boolean)
[16:07:46] [Server thread/DEBUG]: #1 C:/Users/User/Documents/GitHub/pocketmine-mp/src/pocketmine/Player(3560): pocketmine\entity\Human->saveNBT(boolean)
[16:07:46] [Server thread/DEBUG]: #2 C:/Users/User/Documents/GitHub/pocketmine-mp/src/pocketmine/Player(3475): pocketmine\Player->save(boolean)
[16:07:46] [Server thread/DEBUG]: #3 C:/Users/User/Documents/GitHub/pocketmine-mp/src/pocketmine/Player(1976): pocketmine\Player->close(string , string must accept resource packs to join, boolean 1)
[16:07:46] [Server thread/DEBUG]: #4 C:/Users/User/Documents/GitHub/pocketmine-mp/src/pocketmine/network/mcpe/protocol/ResourcePackClientResponsePacket(58): pocketmine\Player->handleResourcePackClientResponse(pocketmine\network\mcpe\protocol\ResourcePackClientResponsePacket object)
[16:07:46] [Server thread/DEBUG]: #5 C:/Users/User/Documents/GitHub/pocketmine-mp/src/pocketmine/Player(3327): pocketmine\network\mcpe\protocol\ResourcePackClientResponsePacket->handle(pocketmine\Player Player())
[16:07:46] [Server thread/DEBUG]: #6 C:/Users/User/Documents/GitHub/pocketmine-mp/src/pocketmine/network/mcpe/protocol/BatchPacket(74): pocketmine\Player->handleDataPacket(pocketmine\network\mcpe\protocol\ResourcePackClientResponsePacket object)
[16:07:46] [Server thread/DEBUG]: #7 C:/Users/User/Documents/GitHub/pocketmine-mp/src/pocketmine/Player(3327): pocketmine\network\mcpe\protocol\BatchPacket->handle(pocketmine\Player Player())
[16:07:46] [Server thread/DEBUG]: #8 C:/Users/User/Documents/GitHub/pocketmine-mp/src/pocketmine/network/mcpe/RakLibInterface(135): pocketmine\Player->handleDataPacket(pocketmine\network\mcpe\protocol\BatchPacket object)
[16:07:46] [Server thread/DEBUG]: #9 C:/Users/User/Documents/GitHub/pocketmine-mp/src/raklib/server/ServerHandler(96): pocketmine\network\mcpe\RakLibInterface->handleEncapsulated(string 192.168.0.7:19132, raklib\protocol\EncapsulatedPacket `............x.c.dd`...@.., integer 0)
[16:07:46] [Server thread/DEBUG]: #10 C:/Users/User/Documents/GitHub/pocketmine-mp/src/pocketmine/network/mcpe/RakLibInterface(78): raklib\server\ServerHandler->handlePacket(boolean)
[16:07:46] [Server thread/DEBUG]: #11 C:/Users/User/Documents/GitHub/pocketmine-mp/src/pocketmine/network/Network(170): pocketmine\network\mcpe\RakLibInterface->process(boolean)
[16:07:46] [Server thread/DEBUG]: #12 C:/Users/User/Documents/GitHub/pocketmine-mp/src/pocketmine/Server(2364): pocketmine\network\Network->processInterfaces(boolean)
[16:07:46] [Server thread/DEBUG]: #13 C:/Users/User/Documents/GitHub/pocketmine-mp/src/pocketmine/Server(2138): pocketmine\Server->tick(boolean)
[16:07:46] [Server thread/DEBUG]: #14 C:/Users/User/Documents/GitHub/pocketmine-mp/src/pocketmine/Server(2021): pocketmine\Server->tickProcessor(boolean)
[16:07:46] [Server thread/DEBUG]: #15 C:/Users/User/Documents/GitHub/pocketmine-mp/src/pocketmine/Server(1642): pocketmine\Server->start(boolean)
[16:07:46] [Server thread/DEBUG]: #16 C:/Users/User/Documents/GitHub/pocketmine-mp/src/pocketmine/PocketMine(493): pocketmine\Server->__construct(BaseClassLoader object, pocketmine\utils\MainLogger object, string C:\Users\User\documents\github\pocketmine-mp\, string C:\Users\User\documents\github\pocketmine-mp\, string C:\Users\User\documents\github\pocketmine-mp\plugins\)
```

**Comment by dktapps** [comment]
Relates to #465 (I don't call parent constructor until after completing login sequence due to the constructor causing packets to be sent at a bad time, causing a client crash (HACK TODO FIX))

**Comment by dktapps** [comment]
Resolved as of ca7ca0cea9468e47d5cc7d1a632400bb234ca05d
